controllers/etcd_custodian_controller.go

```diff
@@ -80,12 +125,183 @@ func (ec *EtcdCustodian) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 	logger := ec.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String())
 
-	// TODO: (timuthy) remove this as it could block important health checks
-	if etcd.Status.LastError != nil && *etcd.Status.LastError != "" {
-		logger.Info(fmt.Sprintf("Requeue item because of last error: %v", *etcd.Status.LastError))
-		return ctrl.Result{
-			RequeueAfter: 30 * time.Second,
-		}, nil
+	// Get or Create full_snapshot_revisions lease object that will help to set BackupReady condition
+	nsName := types.NamespacedName{
+		Name:      "full-snapshot-revisions",
+		Namespace: etcd.Namespace,
+	}
+
+	fullLease := &coordinationv1.Lease{}
```

Comment:
I prefer to create the `Lease` objects in the `etcd-controller` because the objects themselves are bound to the lifecycle of the etcd cluster which is created/updated/deleted by the `etcd-controller`, not by the `custodian`.

Reply:
Agreed